core/CronArchive/SharedSiteIds.php

```diff
@@ -162,6 +176,13 @@ public function getNextSiteId()
             return null;
         }
 
+        if ($this->initialResetQueueTime !== $this->getResetQueueTime()) {
+            // queue was reset/finished by some other process
+            $this->currentSiteId = null;
+            $this->done = true;
+            return null;
+        }
```

Comment:
Should we maybe log here? Maybe a debug log or an info, since it seems like it might be useful to have in CLI output...

Reply:
Not sure it's needed as we would simply stop once iterated over all sites? Not sure how to add easily...

Reply:
I mean a quick log, like, `The shared site ID queue was reset, stopping.` (or whatever actually happens if not stopping). Might be useful when looking through the logs?

Reply:
@diosmosis added a debug log message. It should usually not be that important since it's a regular/normal thing maybe

Reply:
better to use LoggerInterface instead, but otherwise looks good